### ISSUE TITLE
`chroma` is trained to predict the previous token, not the next one

### ISSUE THREAD

**Issue by p3zo** [p3zo/rhythmic-relationships#46]
`chroma/utils.py` returns the pair in one order and `chroma/train.py` reads it in the other:

```python
# utils.py
def parse_batch(batch, device):
    yb = batch
    yb_shifted = torch.roll(yb, 1)          # start-seeded, shifted right
    yb_shifted[:, 0] = ...
    return yb.to(device), yb_shifted.to(device)     # (original, shifted)

# train.py:60
src, tgt = parse_batch(batch, device)       # src = original, tgt = shifted
logits = model(src)
loss = compute_loss(logits=logits, y=tgt, loss_fn=loss_fn)
```

So the decoder is fed the **unshifted** sequence and trained to predict the **shifted** one — at position `t` it sees `y[t]` and is asked for `y[t-1]`. That target is already inside its own causal context, so the loss collapses toward zero while the model learns nothing about predicting a next step.

`hits_encdec` and `encdec` get this right: their `parse_batch` returns `(src, ctx, tgt)` with `ctx` the shifted target, and `model(src, ctx)` is scored against `tgt`.

The mismatch also means training and inference disagree. `chroma/eval.py:inference` generates autoregressively — appends a sampled token, re-runs, takes `logits[:, -1, :]` — which assumes next-token semantics the model was never trained for. So an apparently well-converged chroma run should generate badly, and the low training loss won't show it.

Same shape of bug is latent in `hits_dec_xformer`, whose train loop calls `loss = model(batch)` on a `PartDataset` batch with no shifting at all (#7).

**Fix**: return `(yb_shifted, yb)` so the caller's `src` is the shifted context and `tgt` is what it should predict. Worth an assertion or a test pinning the direction, since the two sequences have identical shapes and nothing catches the swap.
